go/test/endtoend/multipooler_client.go

```diff
@@ -144,6 +144,50 @@ func IsPrimary(addr string) (bool, error) {
 	return resp.Status.PoolerType == clustermetadatapb.PoolerType_PRIMARY, nil
 }
 
+// WaitForPoolerTypeAssigned waits for the pooler type to be assigned (either PRIMARY or REPLICA, not UNKNOWN).
+// This is useful when the test needs to call RPCs that require the pooler type to be known.
+func WaitForPoolerTypeAssigned(t *testing.T, addr string, timeout time.Duration) (clustermetadatapb.PoolerType, error) {
+	t.Helper()
+
+	conn, err := grpc.NewClient(addr, grpccommon.LocalClientDialOptions()...)
+	if err != nil {
+		return clustermetadatapb.PoolerType_UNKNOWN, fmt.Errorf("failed to create client: %w", err)
+	}
+	defer conn.Close()
+
+	client := multipoolermanagerpb.NewMultiPoolerManagerClient(conn)
+	deadline := time.Now().Add(timeout)
+
+	for time.Now().Before(deadline) {
```

Comment:
Feels like a lot of this is duplicating what `require.Eventually()` or `assert.Eventually()` already handle